src/models/transaction.ts

```diff
@@ -368,6 +366,17 @@ class Transaction {
   isBlock(): boolean {
     return this.version === BLOCK_VERSION || this.version === MERGED_MINED_BLOCK_VERSION;
   }
+
+  /**
+   * Set tx timestamp and weight
+   *
+   * @memberof Transaction
+   * @inner
+   */
+  prepareToSend() {
+    this.timestamp = Math.floor(Date.now() / 1000);
```

Comment:
Maybe move to a separate method (e.g. `updateTimestamp()`) and call it from here?